Tests/AdyenTests/Adyen Tests/Analytics/AnalyticsProviderTests.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  AnalyticsProviderTests.swift
+//  Adyen
+//
+//  Created by Naufal Aros on 4/11/22.
+//  Copyright © 2022 Adyen. All rights reserved.
+//
+
+import XCTest
+@testable import Adyen
+@testable import AdyenNetworking
+
+class AnalyticsProviderTests: XCTestCase {
+
+    var apiClient: APIClientMock!
+    var analyticsConfiguration: AnalyticsConfiguration!
+    var sut: AnalyticsProvider!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        apiClient = APIClientMock()
+        analyticsConfiguration = .init()
+        sut = AnalyticsProvider(apiClient: apiClient, configuration: analyticsConfiguration)
+    }
+
+    override func tearDownWithError() throws {
+        apiClient = nil
+        analyticsConfiguration = nil
+        sut = nil
+        try super.tearDownWithError()
+    }
+
+    func testAnalyticsProviderIsInitializedWithCorrectDefaultConfigurationValues() throws {
+        // Given
+        analyticsConfiguration = .init()
+
+        // Then
+        XCTAssertTrue(sut.configuration.isCheckoutAttemptIdEnabled)
+        XCTAssertTrue(sut.configuration.isTelemetryEnabled)
+    }
+
+    func testFetchCheckoutAttemptIdWhenCheckoutAttemptIdIsDisabledShouldCallCompletionWithNilValue() throws {
+        // Given
+        analyticsConfiguration.isCheckoutAttemptIdEnabled = false
+
+        // When
+        sut.fetchCheckoutAttemptId { receivedCheckoutAttemptId in
+
+            // Then
+            XCTAssertNil(receivedCheckoutAttemptId, "The checkoutAttemptId is not nil.")
```

Comment:
I think we should always put `expectation` and fulfill() even if it's not really async.
I had a similar issue and not having expectations can mess `XCTest`

Reply:
All right! Gonna add them to the ones that are missing it. Thanks Eren!